src/commandHandler.ts

```diff
@@ -18,13 +18,13 @@
 import * as vscode from 'vscode';
 import * as path from 'path';
 import {getValue} from './utils';
+import * as extensionFunctions from './extensionFunctions';
```

Comment:
why this one has been chnaged to `import * as from` but not `import {isDefaultNotificationDisabled} from './utils';` What's the difference?

Reply:
I am not sure I understand that question. You are comparing the use of 2 different imports?

Reply:
yes.
Why in a case it has been changed from `import {something} from './myfile'` to `import * as something from './myfile'` and in another case it has not been changed?

Reply:
Can you point me to a specific example? I usually use the import * as when I need more stuff from that file. If its just 1 or 2 then one could argue to import it directly. Though I think that hiding the origin of a function or variable is not a nice thing. I prefer to have the full referencing style.

Reply:
the reference is the pointed code...
i'm just pointing that the way to write doesn't seem coherent. It has been changed in the Pull Request. I'm wondering what is the rule that implied the change for one case and not for the other.

Reply:
I only added one import and did not change another in this file. As already stated before I like the generic style of importing everything more than importing specific functions / variables because that way its clear where the stuff originates. So my clear preference is import * as xyz from.

Reply:
That said I did not change the other ocurances because it wasn't a required thing.

Reply:
So this change is unrelated to ESlint rules then?

Reply:
It was a required change that was unrelated to ESLint itself but unavoidable.

Reply:
Can you explain why it was unavoidable?

Reply:
Because I needed to call methods from extensionFunctions.ts?

Reply:
seems not. There is no modified usage of extensionFunction in the file

Reply:
What are you trying to tell me? That I do not need that import?

Reply:
No. That I don't understand the reason of the modification. The last provided argument seems invalid.

Reply:
Ok, so let me explain it in a bit more detail as it seems you did not look at extensionFunctions at all yet. This file was previously declaring a namespace which was marked by ESLint as a bad thing. If you want to have the rule of this then open up the file from master and your ESLint will show it to you. So the original state of the import in commandHandler.ts was

`import {extensionFunctions} from './extensionFunctions';`

This means we imported the namespace exported from extensionFunctions.ts. This namespace was no longer there after my rework so I had to change the import statement to reflect that change. I hope this is now more clear to you.

Reply:
Too hard to guess that it was coming from modifications in `extensionFunction` as there are a lot of other modifications in the same commit/PR.

Can you precise the rule that says declaring a namespace is a bad thing? Is it this one https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-namespace.md ?
if yes, it is suggesting to use `module` instead. Not to remove it completely. if I understand well, it will keep some code hierarchy and avoids a lot of modifications.

Reply:
The changes I did to the extensionFunction.ts are working fine and all the tests pass too. I also did test them manually as well as Brian did on his side. There was imo no need to use `namespace` at all and that's why I fully removed it. If you want to reintroduce `module` then you are free to submit a PR once this one has been merged.

Reply:
trying to summarize:
- ESlint rule https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-namespace.md suggests to not use namepace anymore
- it suggests to use `module` instead
- you prefer to not follow ESlint recommendation. Id est to remove the namespace/module completely as you think that there is no need of modularity. Even if this PR is about fixing ESlint errors and it implies to modify thousands of line of code instead of a dozen.
- it implies to modify also the import in this other file. And you choose to use a `import * as something from './myfile'` instead of an `import {something1, something2, ... } from './myfile'` even if the second is used by some other imports in the same file. This is to make it clearer from where the stuff originates

I still think that following the Eslint rule recommendation in a PR meant to fix ESLint issues would have been better.

Reply:
Yeah, provide a working PR for it and will happily approve it.